backend/src/main/java/ai/verta/modeldb/dataset/DatasetServiceImpl.java

```diff
@@ -694,16 +698,37 @@ public void deleteDatasets(
         throw new InvalidArgumentException(ModelDBMessages.DATASET_ID_NOT_FOUND_IN_REQUEST);
       }
 
+      Map<String, Long> workspaceIdByDatasetId =
+          request.getIdsList().stream()
+              .collect(
+                  Collectors.toMap(
+                      id -> id,
+                      id ->
+                          roleService
+                              .getEntityResource(id, ModelDBServiceResourceTypes.DATASET)
+                              .getWorkspaceId()));
       deleteRepositoriesByDatasetIds(request.getIdsList());
       DeleteDatasets.Response response =
           DeleteDatasets.Response.newBuilder().setStatus(true).build();
-      saveAuditLogs(
-          Optional.empty(),
-          ModelDBServiceActions.DELETE,
-          request.getIdsList(),
-          ModelDBUtils.getStringFromProtoObject(request),
-          ModelDBUtils.getStringFromProtoObject(response),
-          authService.getWorkspaceIdFromUserInfo(authService.getCurrentLoginUserInfo()));
+      List<AuditLogLocalEntity> auditLogLocalEntities =
+          workspaceIdByDatasetId.entrySet().stream()
+              .map(
+                  entry ->
+                      new AuditLogLocalEntity(
+                          SERVICE_NAME,
+                          authService.getVertaIdFromUserInfo(authService.getCurrentLoginUserInfo()),
+                          ModelDBServiceActions.DELETE,
+                          entry.getKey(),
+                          ModelDBServiceResourceTypes.DATASET,
+                          Service.MODELDB_SERVICE,
+                          MonitoringInterceptor.METHOD_NAME.get(),
+                          ModelDBUtils.getStringFromProtoObjectSilent(request),
+                          ModelDBUtils.getStringFromProtoObjectSilent(response),
+                          entry.getValue()))
```

Comment:
Splitting this seems like an anti-pattern to me. Can we use the same function to avoid divergence?

Reply:
ok, used the same